R/setkey.R

```diff
@@ -353,6 +354,9 @@ CJ <- function(..., sorted = TRUE, unique = FALSE)
                 l[[i]] = rep.int(rep.int(y, times = rep.int(x[i], n[i])), times = nrow/(x[i]*n[i]))
             if (any(class(l[[i]]) != j[[i]]))
                 setattr(l[[i]], 'class', j[[i]]) # reset "Date" class - rep.int coerces to integer
+            if (any(!sapply(tzones, is.null))) {
```

Comment:
I suppose this could make sense if `is.null` is significantly faster than just setting the attributes even if they're `NULL`? In any case, the same would go for the lines that reset class immediately above?